PolyType type variable whose lower bound is another type variable of the PolyType is unsatisfiable

The following code does not compile with dotty but works with scalac:
```scala
class X
class A {
  def foo[T1, T2 >: T1] = {}
  def foo2[T1, T2 <: T1] = {}

  def test = {
    foo[X, X] // breaks
    foo2[X, X] // works
  }
}
```
```scala
[error] try/tvar.scala:6: error: Type argument X does not conform to lower bound T1 
[error]     foo[X, X] // breaks
[error]            ^
[error] one error found
```

--- a/src/dotty/tools/dotc/core/TypeOps.scala
+++ b/src/dotty/tools/dotc/core/TypeOps.scala
@@ -352,10 +352,11 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
         //println(i"instantiating ${bounds.hi} with $argTypes")
         //println(i" = ${instantiate(bounds.hi, argTypes)}")
         val hiBound = instantiate(bounds.hi, argTypes.mapConserve(_.bounds.hi))
+        val loBound = instantiate(bounds.lo, argTypes.mapConserve(_.bounds.lo))
           // Note that argTypes can contain a TypeBounds type for arguments that are
           // not fully determined. In that case we need to check against the hi bound of the argument.
         if (!(lo <:< hiBound)) violations += ((arg, "upper", hiBound))
-        if (!(bounds.lo <:< hi)) violations += ((arg, "lower", bounds.lo))
+        if (!(loBound <:< hi)) violations += ((arg, "lower", bounds.lo))
       }
       arg.tpe match {
         case TypeBounds(lo, hi) => checkOverlapsBounds(lo, hi)

--- a/tests/pos/i523.scala
+++ b/tests/pos/i523.scala
@@ -1,0 +1,8 @@
+class X
+class A {
+  def foo[T1, T2 >: T1]: Unit = {}
+
+  def test = {
+    foo[X, X]
+  }
+}

